### PR TITLE
[DROOLS-5490] Implement built-in functions scheleton management

### PR DESCRIPTION
**Pull request by gitgabrio**
@danielezonca @kostola  @jiripetrlik

See https://issues.redhat.com/browse/DROOLS-5490

Minor refactoring of ExpressionUtils

### CODE REVIEW

File: kie-pmml-trusty/kie-pmml-api/src/main/java/org/kie/pmml/api/enums/OPERATOR.java [hunk 1]
```diff
@@ -43,6 +43,14 @@
                                                                  LESS_OR_EQUAL,
                                                                  GREATER_THAN,
                                                                  GREATER_OR_EQUAL);
+
+    /**
+     * <code>OPERATOR</code>s that operates <b>ONLY</b >with <code>Number</code>s
+     */
+    static final List<OPERATOR> ONLY_NUMBER_OPERATORS = Arrays.asList(LESS_THAN,
```
Comment (danielezonca) [inline kie-pmml-trusty/kie-pmml-api/src/main/java/org/kie/pmml/api/enums/OPERATOR.java:50]: What about `EQUAL` and `NOT_EQUAL`?

**Comment by gitgabrio** [comment]
Jenkins retest this

File: kie-pmml-trusty/kie-pmml-api/src/main/java/org/kie/pmml/api/enums/BUILTIN_FUNCTIONS.java [hunk 2]
```diff
@@ -117,23 +118,143 @@ public String getName() {
...
+                return sum(inputData);
+            case UPPERCASE:
+                return uppercase(inputData);
             default:
                 throw new KiePMMLException("Unmanaged BUILTIN_FUNCTIONS " + this);
         }
     }
 
+    private double avg(final Object[] inputData) {
+        checkNumbers(inputData, inputData.length);
```
Comment (jiripetrlik) [inline kie-pmml-trusty/kie-pmml-api/src/main/java/org/kie/pmml/api/enums/BUILTIN_FUNCTIONS.java:151]: Do we check somewhere that the inputData array is not empty?

File: kie-pmml-trusty/kie-pmml-api/src/main/java/org/kie/pmml/api/enums/BUILTIN_FUNCTIONS.java [hunk 3]
```diff
@@ -117,23 +118,143 @@ public String getName() {
...
+    }
+
+    private String lowercase(final Object[] inputData) {
+        checkStrings(inputData, 1);
+        return ((String) inputData[0]).toLowerCase();
+    }
+
+    private double max(final Object[] inputData) {
+        checkNumbers(inputData, inputData.length);
+        return Arrays.stream(inputData)
```
Comment (jiripetrlik) [inline kie-pmml-trusty/kie-pmml-api/src/main/java/org/kie/pmml/api/enums/BUILTIN_FUNCTIONS.java:172]: Do we check somewhere that the inputData array is not empty?

File: kie-pmml-trusty/kie-pmml-api/src/main/java/org/kie/pmml/api/enums/BUILTIN_FUNCTIONS.java [hunk 4]
```diff
@@ -117,23 +118,143 @@ public String getName() {
...
+                .mapToDouble(num -> ((Number)num).doubleValue())
+                .max()
+                .orElseThrow(() -> new KieEnumException("Failed to find maximum value"));
+    }
+
+    private double median(final Object[] inputData) {
+        checkNumbers(inputData, inputData.length);
```
Comment (jiripetrlik) [inline kie-pmml-trusty/kie-pmml-api/src/main/java/org/kie/pmml/api/enums/BUILTIN_FUNCTIONS.java:179]: Do we check somewhere that the inputData array is not empty?

File: kie-pmml-trusty/kie-pmml-api/src/main/java/org/kie/pmml/api/enums/BUILTIN_FUNCTIONS.java [hunk 5]
```diff
@@ -117,23 +118,143 @@ public String getName() {
...
+                .mapToDouble(num -> ((Number)num).doubleValue())
+                .sorted();
+        OptionalDouble toReturn = inputData.length % 2 == 0 ?
+                sortedValues.skip(inputData.length / 2 - (long)1).limit(2).average() :
+                sortedValues.skip(inputData.length / 2).findFirst();
+        return toReturn.orElseThrow(() -> new KieEnumException("Failed to find median value"));
+    }
+
+    private double min(final Object[] inputData) {
+        checkNumbers(inputData, inputData.length);
```
Comment (jiripetrlik) [inline kie-pmml-trusty/kie-pmml-api/src/main/java/org/kie/pmml/api/enums/BUILTIN_FUNCTIONS.java:190]: Do we check somewhere that the inputData array is not empty?

File: kie-pmml-trusty/kie-pmml-api/src/main/java/org/kie/pmml/api/enums/BUILTIN_FUNCTIONS.java (hunk 4 above)
Comment (gitgabrio) [reply kie-pmml-trusty/kie-pmml-api/src/main/java/org/kie/pmml/api/enums/BUILTIN_FUNCTIONS.java:179]: @jiripetrlik 
Done

File: kie-pmml-trusty/kie-pmml-api/src/main/java/org/kie/pmml/api/enums/BUILTIN_FUNCTIONS.java (hunk 5 above)
Comment (gitgabrio) [reply kie-pmml-trusty/kie-pmml-api/src/main/java/org/kie/pmml/api/enums/BUILTIN_FUNCTIONS.java:190]: @jiripetrlik 
Done

File: kie-pmml-trusty/kie-pmml-api/src/main/java/org/kie/pmml/api/enums/BUILTIN_FUNCTIONS.java (hunk 3 above)
Comment (gitgabrio) [reply kie-pmml-trusty/kie-pmml-api/src/main/java/org/kie/pmml/api/enums/BUILTIN_FUNCTIONS.java:172]: @jiripetrlik 
Done

File: kie-pmml-trusty/kie-pmml-api/src/main/java/org/kie/pmml/api/enums/BUILTIN_FUNCTIONS.java (hunk 2 above)
Comment (gitgabrio) [reply kie-pmml-trusty/kie-pmml-api/src/main/java/org/kie/pmml/api/enums/BUILTIN_FUNCTIONS.java:151]: @jiripetrlik 
Done

**Review by danielezonca — APPROVED** [review]

File: kie-pmml-trusty/kie-pmml-api/src/main/java/org/kie/pmml/api/enums/OPERATOR.java (hunk 1 above)
Comment (gitgabrio) [reply kie-pmml-trusty/kie-pmml-api/src/main/java/org/kie/pmml/api/enums/OPERATOR.java:50]: <code>OPERATOR</code>s that operates <b>ONLY</b >with <code>Number</code>s

**Review by jiripetrlik — APPROVED** [review]
@gitgabrio 
Thank you for adding additional check. LGTM

**Pull request merged**